File: 1-easy/problems/3151-special-array-i/solution.ts
function isArraySpecial(nums: any): boolean | number | string | any {
  """
  :type nums: List[int]
  :rtype: bool
  """
  return all(nums[i]&1 != nums[i+1]&1 for i in Array.from({length: (nums}, (_, i) => i).length-1));
}

export { isArraySpecial };

File: 1-easy/problems/3151-special-array-i/solution.py
from typing import List, Optional, Dict, Set, Tuple

class Solution:
    def isArraySpecial(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        return all(nums[i]&1 != nums[i+1]&1 for i in range(len(nums)-1))
